tests/unit/amo/components/TestAddonMoreInfo.js

```diff
@@ -512,6 +515,58 @@ describe(__filename, () => {
     expect(statsLink).toHaveLength(1);
   });
 
+  it('render related categories of the addon if existed', () => {
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      categories: {
+        firefox: ['other', 'testing'],
+      },
+    });
+
+    const fakeCategoryState = {
```

Comment:
The approach you are taking here won't work - we need to do something quite different. What we need to do is create data in the redux store that will support our test case. That way, when `mapStateToProps` is executed in `AddonMoreInfo`, it will yield the expected result. This is a bit complex, but I will include a test below that works. We need to do the following things:

1. Create an add-on with the categories we expect, for a specific `clientApp`. You are already doing this, but my code is a bit different.
2. Load the associated categories into the redux state. You are attempting to do something similar here, but, as I said, this won't work. You need to use `loadCategories` instead.
3. Make sure that the `clientApp` for the request matches the one you used for the categories in the `addon`, by using `dispatchClientMetadata`.

Applying these changes to your patch will yield the test I am talking about:
```diff
diff --git a/tests/unit/amo/components/TestAddonMoreInfo.js b/tests/unit/amo/components/TestAddonMoreInfo.js
index 7ca18f864..98bbcfe96 100644
--- a/tests/unit/amo/components/TestAddonMoreInfo.js
+++ b/tests/unit/amo/components/TestAddonMoreInfo.js
@@ -12,7 +12,7 @@ import {
   CLIENT_APP_FIREFOX,
 } from 'core/constants';
 import { createInternalAddon } from 'core/reducers/addons';
-import { getCategoryNames } from 'core/reducers/categories';
+import { loadCategories } from 'core/reducers/categories';
 import { formatFilesize } from 'core/i18n/utils';
 import {
   createContextWithFakeRouter,
@@ -21,6 +21,7 @@ import {
   dispatchClientMetadata,
   dispatchSignInActions,
   fakeAddon,
+  fakeCategory,
   fakeTheme,
   fakeI18n,
   fakeVersion,
@@ -515,56 +519,39 @@ describe(__filename, () => {
     expect(statsLink).toHaveLength(1);
   });

-  it('render related categories of the addon if existed', () => {
+  it('renders related categories of the addon', () => {
+    const category1 = {
+      ...fakeCategory,
+      type: ADDON_TYPE_EXTENSION,
+      name: 'Category 1',
+      slug: 'category-1',
+    };
+    const category2 = {
+      ...fakeCategory,
+      type: ADDON_TYPE_EXTENSION,
+      name: 'Category 2',
+      slug: 'category-2',
+    };
+
     const addon = createInternalAddon({
       ...fakeAddon,
+      type: ADDON_TYPE_EXTENSION,
       categories: {
-        firefox: ['other', 'testing'],
+        [CLIENT_APP_FIREFOX]: [category1.slug, category2.slug],
       },
     });

-    const fakeCategoryState = {
-      firefox: {
-        extension: {
-          other: {
-            id: 1,
-            name: 'Testing category',
-            slug: 'other',
-            application: CLIENT_APP_FIREFOX,
-            misc: false,
-            type: ADDON_TYPE_EXTENSION,
-            weight: 1,
-            description: 'I am a cool category for doing things',
-          },
-          testing: {
-            id: 2,
-            name: 'Another testing category',
-            slug: 'testing',
-            application: CLIENT_APP_FIREFOX,
-            misc: false,
-            type: ADDON_TYPE_EXTENSION,
-            weight: 1,
-            description: 'I am a cool category for doing things',
-          },
-        },
-      },
-    };
-
-    const relatedCategories = getCategoryNames(
-      fakeCategoryState,
-      addon.categories,
-      CLIENT_APP_FIREFOX,
-      addon.type,
-    );
+    dispatchClientMetadata({ clientApp: CLIENT_APP_FIREFOX, store });
+    store.dispatch(loadCategories({ results: [category1, category2] }));

     const root = render({
       addon,
-      relatedCategories,
+      store,
     });

     expect(
       root.find('.AddonMoreInfo-related-categories').children(),
-    ).toHaveText('Testing category, Another testing category');
+    ).toHaveText(`${category1.name}, ${category2.name}`);
   });

   it('links to version history if add-on is extension', () => {
```

Reply:
@bobsilverberg Thanks for the above suggestions. It worked perfectly. Actually, I was trying to test that `getCategoryNames` function too in the same test, but yes it should be in the different test and `categories` should dispatched to the store for this.